model.js

```diff
@@ -1,17 +1,64 @@
+const path = require('path')
+const select = require('./lib/s3')
+const { translate } = require('./lib/translate')
+const config = require('config')
+if (!config.koopProviderS3Select) throw new Error(`ERROR: "koopProviderS3Select" must be defined in your config.`)
+if (!config.koopProviderS3Select.stores) throw new Error(`ERROR: "koopProviderS3Select.stores" must be defined in your config.`)
+const stores = config.koopProviderS3Select.stores
+
 /**
  * Model constructor
  */
 function Model () {}
 
 /**
- * Fetch data from source.  Pass result or error to callback.
- * This is the only public function you need to implement on Model
- * @param {object} express request object
+ * Fetch data from S3. Pass result or error to callback.
+ * @param {object} req express request object
  * @param {function} callback
  */
-Model.prototype.getData = function (req, callback) {
-  // stub
-  callback(null, {})
+Model.prototype.getData = async function (req, callback) {
```

Comment:
should we set some metadata annotations here?

e.g. `filtersApplied.where`

Reply:
I guess we can only do that after we use custom sql. Should just add a TODO on that